pkg/apis/apps/validation/validation.go

```diff
@@ -128,6 +130,12 @@ func ValidateStatefulSetSpec(spec *apps.StatefulSetSpec, fldPath *field.Path, op
 
 	allErrs = append(allErrs, apivalidation.ValidateNonnegativeField(int64(spec.Replicas), fldPath.Child("replicas"))...)
 	allErrs = append(allErrs, apivalidation.ValidateNonnegativeField(int64(spec.MinReadySeconds), fldPath.Child("minReadySeconds"))...)
+	if utilfeature.DefaultFeatureGate.Enabled(features.StatefulSetStartOrdinal) {
+		if spec.Ordinals != nil {
+			replicaStartOrdinal := spec.Ordinals.Start
```

Comment:
I'd also suggest adding a check for start < replicas.

Reply:
Start and replicas are effectively independent, so I think the only validation we can do is to ensure that start is non-negative. Consider the two scenarios:

start = 0, replicas = 3. Pods are named (pod-0, pod-1, pod-2)

start = 3, replicas = 2. Pods are named (pod-3, pod-4)

Reply:
Good point, I forgot about that use-case, double checked it's explicitly mentioned in the KEP.